packages/eas-cli/src/build/__tests__/handleBuildRequestError-test.ts

```diff
@@ -131,6 +132,26 @@ describe(Build.name, () => {
               assertReThrownError(caughtError as Error, TurtleDeprecatedJobFormatError, 'Error 1');
             }
           });
+
+          it('throws correct EasBuildResourceClassNotAvailableInFreeTierError', async () => {
+            const platform = Platform.ANDROID;
+            const graphQLError = getGraphQLError(
+              'Error 1',
+              'EAS_BUILD_RESOURCE_CLASS_NOT_AVAILABLE_IN_FREE_TIER'
+            );
+            const graphQLErrors = [graphQLError];
+            const error = new CombinedError({ graphQLErrors });
+
+            try {
+              handleBuildRequestError(error, platform);
+            } catch (caughtError) {
+              assertReThrownError(
+                caughtError as Error,
+                EasBuildResourceClassNotAvailableInFreeTierError,
+                'Error 1'
+              );
+            }
```

Comment:
I think this could be cleaner. The problem with this approach is that if `handleBuildRequestError` doesn't throw an error, this test doesn't check anything. 
See the approach that I used in `@expo/steps` and refactor this test suite if it's not too much trouble for you. (It's also described on Stackoverflow, see under "Edit 2022" - https://stackoverflow.com/questions/46042613/how-to-test-the-type-of-a-thrown-exception-in-jest)

- Util: https://github.com/expo/eas-build/blob/main/packages/steps/src/__tests__/utils/error.ts
- Usage: https://github.com/expo/eas-build/blob/4f86863e35b2400ab9ef91141f28ef39e101daa6/packages/steps/src/__tests__/BuildStepContext-test.ts#L69